File: android/src/main/kotlin/com/pixmerc/fl_env/storage/RuntimeStorage.kt
package com.pixmerc.fl_env.storage

import android.content.Context
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKey

/**
 * Persists fl_env runtime state using EncryptedSharedPreferences.
 *
 * Phase 1 stores only the active tier name. Phase 2 will add per-tier
 * caching and keychain-backed key rotation state.
 */
internal class RuntimeStorage(context: Context) {

    private val prefs by lazy {
        val masterKey = MasterKey.Builder(context)
            .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
            .build()

        EncryptedSharedPreferences.create(
            context,
            "fl_env_prefs",
            masterKey,
            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM,
        )
    }

    fun getActiveTier(): String? = prefs.getString(KEY_ACTIVE_TIER, null)

    fun setActiveTier(tier: String) {
        prefs.edit().putString(KEY_ACTIVE_TIER, tier).apply()
    }

    private companion object {
        const val KEY_ACTIVE_TIER = "active_tier"
    }
}

File: ios/Classes/Storage/RuntimeStorage.swift
import Foundation

/// Persists fl_env runtime state using UserDefaults.
///
/// Phase 1 stores only the active tier name. Phase 2 will migrate to
/// Keychain-backed storage for sensitive runtime state.
final class RuntimeStorage {

    static let shared = RuntimeStorage()
    private init() {}

    private let defaults = UserDefaults.standard
    private let activeTierKey = "fl_env_active_tier"

    func getActiveTier() -> String? {
        return defaults.string(forKey: activeTierKey)
    }

    func setActiveTier(_ tier: String) {
        defaults.set(tier, forKey: activeTierKey)
    }
}
